### ISSUE TITLE
Do not bind or warn about files in the state file that do not exist in in the config anymore

### ISSUE THREAD

**Issue by kris-watts-gravwell** [gravwell/gravwell#1016]
## What are the steps to reproduce this issue?
1. add a few file watch patterns
2. let file follower start
3. shutdown file follower
4. remove a watcher
5. remove the file that was being watched
6. start the file follower

## What happens?

1. File follower will bounce up and complain that it can't watch a file

## What were you expecting to happen?

do not not even bark about the file

## Any logs, error output, etc?

```
<11>1 2024-02-22T20:20:01.123456Z foo.bar.baz filefollow - filewatch/filewatch.go:523 [gw@1 path="/var/log/loggy.log.1" error="no such file or directory"] failed to watch file
```

## Any other comments?

I think we should scan the state file and eject anything in the state file that doesn't match something in the config, maybe log that we are ejecting a file that we used to.

## What versions of software are you using?

5.4.5-2

**Comment by john-floren-gravwell** [comment]
I defined these followers:

```
[Follower "foo"]
	Base-Directory="/tmp/foo"
	File-Filter="*"
	Tag-Name=foo
	Assume-Local-Timezone=true #Default for assume localtime is false

[Follower "bar"]
	Base-Directory="/tmp/bar"
	File-Filter="*" 
	Tag-Name=bar
	Assume-Local-Timezone=true #Default for assume localtime is false
```

I created /tmp/foo and /tmp/bar, then created files in both directories.

I tried various combinations of removing followers and removing files, and was unable to get any sort of error out of the thing.

Note also that the error message you've pasted shows up when we saw a *write* to a file we should be watching, but then the attempt to start a follower on that file failed. When I see the ".1" suffix in that filename, I start to suspect that we raced a log rotation and lost in some way.
